src/vmm/src/builder.rs

```diff
@@ -425,7 +359,32 @@ pub fn build_microvm_for_boot(
         track_dirty_pages,
         vm_resources.vm_config().vcpu_count,
     )?;
-    let vcpu_config = vm_resources.custom_vcpu_config(vmm.vm.guest_cpu_config.clone());
+
+    #[cfg(target_arch = "aarch64")]
+    let custom_cpu_config = if let Some(ref cpu_template) = vm_resources.vm_config().cpu_template {
+        let vcpu = vcpus
+            .get(0)
+            .expect("There should always be at least 1 vcpu");
+
+        Some(
+            CpuConfiguration::from_cpu_template(&vcpu.kvm_vcpu.fd, cpu_template)
+                .map_err(ApplyCpuTemplate)?,
+        )
+    } else {
+        None
+    };
+
+    #[cfg(target_arch = "x86_64")]
+    let custom_cpu_config = if let Some(ref cpu_template) = vm_resources.vm_config().cpu_template {
+        CpuConfiguration::from_cpu_template(&vmm.vm, cpu_template).map_err(ApplyCpuTemplate)?
```

Comment:
Having a different API for the architectures here doesn't quite make sense from the caller's point of view. The configuration between the two is understandably different internally, but should require only the same information/references passed to it when it comes time to use the template. This makes sense for the caller of the function so there isn't differentiation at that level, but also makes the unit tests easier to manage as well.